### PR TITLE
Gifify local videos

### PR DESCRIPTION
**Pull request by KevinLiddle**
* Bumps to ruby 2.2.2
* (Hopefully) fixes the empty list error by not passing audio/video codec options to the FFMPEG transcoding
* Adds support for converting local videos to GIFs
* Tested converting from the following formats:
  * MOV
  * FLV
  * WMV
  * WEBM
  * AVI
  * MP4

Please pull it down, try it out, make sure the tests pass and it works on a machine that's not mine

### CODE REVIEW

**Comment by KevinLiddle** [comment]
It looks like the 1.9.3 build timed out when installing gems, but when I ran the tests locally under 1.9.3 I got the regression tests failing. It wasn't able to download the videos from youtube, getting an EOFError and a "Connection reset by peer" error. Not sure what the problem is there.

**Comment by ecmendenhall** [comment]
Yeah, just a timeout. CI runs and end-to-end integration spec, so it happens sometimes.

File: bin/wgif [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env ruby
 
+$: << File.join(File.expand_path(__dir__), "..", "lib")
```
Comment (ecmendenhall) [inline bin/wgif]: Two things here: 1) `__dir__` is Ruby 2+ only, 2) I think the `".."` is incorrect (although I can never decipher these :unamused:):

```
irb(main):001:0> RUBY_VERSION
=> "2.2.2"
irb(main):002:0> File.join(File.expand_path(__dir__), "..", "lib")
=> "/Users/ecm/Projects/ruby/wgif/../lib"
```

If you want to support my arbitrarily selected assortment of Ruby versions, I think you need `File.join(File.expand_path('..', __FILE__), 'lib')`:

```
irb(main):001:0> RUBY_VERSION
=> "2.2.2"
irb(main):002:0> File.join(File.expand_path('..', __FILE__), 'lib')
=> "/Users/ecm/Projects/ruby/wgif/lib"
```

```
irb(main):001:0> RUBY_VERSION
=> "2.0.0"
irb(main):002:0> File.join(File.expand_path('..', __FILE__), 'lib')
=> "/Users/ecm/Projects/ruby/wgif/lib"
```

```
irb(main):001:0> RUBY_VERSION
=> "1.9.3"
irb(main):002:0> File.join(File.expand_path('..', __FILE__), 'lib')
=> "/Users/ecm/Projects/ruby/wgif/lib"
```

Comment (KevinLiddle) [inline bin/wgif]: I'm not even sure I knew `__dir__` was a thing. I think I tried to recreate this line from memory and lucked into `__dir__`. Also not sure how things were working for me with it going to the wrong place. Anyway, fixing now.

**Comment by KevinLiddle** [comment]
@ecmendenhall Updated and passing. Ready for another review.

**Comment by ecmendenhall** [comment]
:+1: :x: :100:

**Pull request merged**